bpf/kprobe_pwru.c

```diff
@@ -66,11 +58,41 @@ struct {
 	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
 } events SEC(".maps");
 
+union v6addr {
+	struct {
+		u32 p1;
+		u32 p2;
+		u32 p3;
+		u32 p4;
+	};
+	struct {
+		u64 d1;
+		u64 d2;
+	};
+	__u8 addr[16];
+} __attribute__((packed));
+
+struct filter_cfg {
+	u32 mark;
+	u8 ipv6;
+	union v6addr saddr;
```

Comment:
Maybe let's make the addr type a union (v4addr and v6addr) for the sake of readability?

Reply:
OK, i will fix it later.